include/flash_map_pm.h

```diff
@@ -44,6 +44,9 @@
 #define FLASH_AREA_SIZE(label) \
 	UTIL_CAT(PM_, UTIL_CAT(UTIL_CAT(PM_, UTIL_CAT(PM_ID(label), _LABEL)), _SIZE))
 
+#define FLASH_AREA_DEVICE(label) \
+	UTIL_CAT(PM_, UTIL_CAT(UTIL_CAT(PM_, UTIL_CAT(PM_ID(label), _LABEL)), _DEV_NAME))
```

Comment:
This is not correct solution, the macro returns string that represents device name; the FLASH_AREA_DEVICE macro returns pointer to a device structure that the area resides on.
This macro should be either changed to invocation of device_get_binding, or additional support and changes in python script are required, to use DTS, to figure out pointers to flash devices that the storage resides on.

Reply:
Added call to devcie_get_binding, nvs sample now works with following patch

```
diff --git a/samples/subsys/nvs/prj.conf b/samples/subsys/nvs/prj.conf
index 2cf7ed9535..b469f97514 100644
--- a/samples/subsys/nvs/prj.conf
+++ b/samples/subsys/nvs/prj.conf
@@ -6,3 +6,5 @@ CONFIG_LOG=y
 CONFIG_NVS_LOG_LEVEL_DBG=y
 CONFIG_REBOOT=y
 CONFIG_MPU_ALLOW_FLASH_WRITE=y
+
+CONFIG_PM_SINGLE_IMAGE=y
```